sky/shell/ui/engine.cc

```diff
@@ -96,27 +96,18 @@ void Engine::Init() {
   blink::initialize(g_platform_impl);
 }
 
-void Engine::BeginFrame(base::TimeTicks frame_time) {
+std::unique_ptr<LayerTree> Engine::BeginFrame(base::TimeTicks frame_time) {
   TRACE_EVENT0("sky", "Engine::BeginFrame");
 
+  std::unique_ptr<LayerTree> layer_tree;
   if (sky_view_)
-    sky_view_->BeginFrame(frame_time);
-}
-
-PassRefPtr<SkPicture> Engine::Paint() {
-  TRACE_EVENT0("sky", "Engine::Paint");
-
-  if (sky_view_) {
-    RefPtr<SkPicture> picture = sky_view_->Paint();
-    if (picture)
-      return picture.release();
-  }
+    layer_tree = sky_view_->BeginFrame(frame_time);
+  if (!layer_tree)
+    layer_tree.reset(new LayerTree());
 
-  SkPictureRecorder recorder;
-  SkCanvas* canvas = recorder.beginRecording(
-      physical_size_.width(), physical_size_.height());
-  canvas->clear(SK_ColorBLACK);
-  return adoptRef(recorder.endRecordingAsPicture());
+  layer_tree->set_frame_size(SkISize::Make(physical_size_.width(),
+                                           physical_size_.height()));
+  return std::move(layer_tree);
```

Comment:
you don't need (or imo want) std::move here.  this is a return expression so layer_tree is an 'xvalue' (for expiring value) which is an rvalue type already.  std::move() is just converting your rvalue into an rvalue. it compiles for the same reason std::move(std::move(std::move(layer_tree))) would, but just confuses the issue